azext_iot/operations/hub.py

```diff
@@ -3301,3 +3303,290 @@ def _customize_device_tracing_output(device_id, desired, reported):
         ):
             output["isSynced"] = True
     return output
+
+
+# Topic Space Preview
+
+
+def iot_hub_topic_space_create(
+    cmd,
+    topic_name,
+    topic_templates,
+    topic_type,
+    hub_name=None,
+    resource_group_name=None,
+    login=None,
+    auth_type_dataplane=None,
+    target=None,
+    service_sdk=None,
+):
+    if not target:
+        discovery = IotHubDiscovery(cmd)
+        target = discovery.get_target(
+            hub_name=hub_name,
+            resource_group_name=resource_group_name,
+            login=login,
+            auth_type=auth_type_dataplane,
+        )
+        resolver = SdkResolver(target=target)
+        service_sdk = resolver.get_sdk(SdkType.service_sdk)
+
+    topic_type = TopicSpaceType(topic_type)
+    if topic_type == TopicSpaceType.HighFanout:
+        raise CLIError("Only LowFanout and PublishOnly topic types are supported right now")
+
+    if topic_templates and looks_like_file(topic_templates[0]):
+        topic_templates = process_json_arg(topic_templates[0], "Topic Space Templates")
+
+    try:
+        return service_sdk.topic_space.put_topic_space(
+            id=topic_name,
+            name=topic_name,
+            properties={
+                "topicspace_type": topic_type,
+                "topic_templates": topic_templates
```

Comment:
What happens if topic_templates is `[]` , does the service handle well or should the client handle?

Reply:
Service handles this well

Reply:
Service does not handle duplicates templates well though